### PR TITLE
[examples] misc fixes

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR does misc updates to dreambooth and TI scripts, mainly

- Fix logging
- add an option to be able to use TF32
- add `report_to` arg to be able to use other loggers than `tensorboard`
- add some more comments and reorganize the code.

### CODE REVIEW

File: examples/dreambooth/train_dreambooth.py [hunk 1]
```diff
@@ -435,9 +457,26 @@ def main(args):
...
 
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        level=logging.INFO,
+    )
+    logger.info(accelerator.state, main_process_only=False)
+    if accelerator.is_local_main_process:
+        datasets.utils.logging.set_verbosity_warning()
+        transformers.utils.logging.set_verbosity_info()
```
Comment (patrickvonplaten) [inline examples/dreambooth/train_dreambooth.py:468]: why set to info? That's a lot of logging messages then no?

File: examples/textual_inversion/textual_inversion.py [hunk 2]
```diff
@@ -484,6 +504,24 @@ def main():
     text_encoder.text_model.final_layer_norm.requires_grad_(False)
     text_encoder.text_model.embeddings.position_embedding.requires_grad_(False)
 
+    if args.gradient_checkpointing:
+        # Keep unet in train mode if we are using gradient checkpointing to save memory.
+        # The dropout is 0 so it doesn't matter if we are in eval or train mode.
```
Comment (patrickvonplaten) [inline examples/textual_inversion/textual_inversion.py:509]:
```suggestion
        # The dropout cannot be != 0 so it doesn't matter if we are in eval or train mode.
```

File: examples/dreambooth/train_dreambooth.py (hunk 1 above)
Comment (patil-suraj) [reply examples/dreambooth/train_dreambooth.py:469]: aah good catch, transformers should be set to warning.

File: examples/dreambooth/train_dreambooth.py [hunk 3]
```diff
@@ -236,6 +240,24 @@ def parse_args(input_args=None):
...
+        ),
+    )
+    parser.add_argument(
+        "--report_to",
+        type=str,
+        default="tensorboard",
+        help=(
+            'The integration to report the results and logs to. Supported platforms are `"tensorboard"`,'
+            ' `"wandb"` and `"comet_ml"`. Use `"all"` (default) to report to all integrations.'
+            "Only applicable when `--with_tracking` is passed."
```
Comment (pcuenca) [inline examples/dreambooth/train_dreambooth.py:258]:
```suggestion
            'The integration to report the results and logs to. Supported platforms are `"tensorboard"`'
            ' (default)`, "wandb"` and `"comet_ml"`. Use `"all"` to report to all integrations.'
```

File: examples/textual_inversion/textual_inversion.py [hunk 4]
```diff
@@ -207,6 +211,24 @@ def parse_args():
...
+        ),
+    )
+    parser.add_argument(
+        "--report_to",
+        type=str,
+        default="tensorboard",
+        help=(
+            'The integration to report the results and logs to. Supported platforms are `"tensorboard"`,'
+            ' `"wandb"` and `"comet_ml"`. Use `"all"` (default) to report to all integrations.'
+            "Only applicable when `--with_tracking` is passed."
```
Comment (pcuenca) [inline examples/textual_inversion/textual_inversion.py:229]:
```suggestion
            'The integration to report the results and logs to. Supported platforms are `"tensorboard"`'
            ' (default)`, "wandb"` and `"comet_ml"`. Use `"all"` to report to all integrations.'
```

**Review by pcuenca — COMMENTED** [review]
Awesome!

**Comment by pcuenca** [comment]
Lol, was too slow to review. I'll open a PR for those minor docstring fixes.

**Pull request merged**